dlpy/tests/test_image_embedding_table.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright SAS Institute
+#
+#  Licensed under the Apache License, Version 2.0 (the License);
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+# NOTE: This test requires a running CAS server.  You must use an ~/.authinfo
+#       file to specify your username and password.  The CAS host and port must
+#       be specified using the CASHOST and CASPORT environment variables.
+#       A specific protocol ('cas', 'http', 'https', or 'auto') can be set using
+#       the CASPROTOCOL environment variable.
+
+import os
+import unittest
+import swat
+import swat.utils.testing as tm
+
+from dlpy.images import ImageTable
+from dlpy.utils import DLPyError
+from dlpy.image_embedding import ImageEmbeddingTable
+
+
+class TestImageEmbeddingTable(unittest.TestCase):
+    # Create a class attribute to hold the cas host type
+    server_type = None
+    s = None
+    server_sep = '/'
+    data_dir = None
+
+    @classmethod
+    def setUpClass(cls):
+        swat.reset_option()
+        swat.options.cas.print_messages = False
+        swat.options.interactive_mode = False
+
+        cls.s = swat.CAS()
+        cls.server_type = tm.get_cas_host_type(cls.s)
+
+        cls.server_sep = '\\'
+        if cls.server_type.startswith("lin") or cls.server_type.startswith("osx"):
+            cls.server_sep = '/'
+
+        if 'DLPY_DATA_DIR' in os.environ:
+            cls.data_dir = os.environ.get('DLPY_DATA_DIR')
+            if cls.data_dir.endswith(cls.server_sep):
+                cls.data_dir = cls.data_dir[:-1]
+            cls.data_dir += cls.server_sep
+
+        if "DLPY_DATA_DIR_LOCAL" in os.environ:
+            cls.local_dir = os.environ.get("DLPY_DATA_DIR_LOCAL")
+
+        # the server path that points to DLPY_DATA_DIR_LOCAL
+        if "DLPY_DATA_DIR_SERVER" in os.environ:
```

Comment:
Not sure what you mean.

Reply:
something like checking if the server path ends with a server_sep do nothing, otherwise add one so that we dont mind if the user pass a path with an ending slash or not.

Reply:
I see. I added the check to server_dir but not local_dir.
I actually do not use these envs for this py file but just added them for future uses.